nixos/modules/services/networking/tayga.nix

```diff
@@ -0,0 +1,221 @@
+          type = types.str;
+          description = lib.mdDoc "IPv${toString v} address.";
+        };
+
+        prefixLength = mkOption {
+          type = types.addCheck types.int (n: n >= 0 && n <= (if v == 4 then 32 else 128));
+          description = lib.mdDoc ''
+            Subnet mask of the interface, specified as the number of
+            bits in the prefix ("${if v == 4 then "24" else "64"}").
+          '';
+        };
+      };
+    };
+
+  versionOpts = v: {
+    options = {
+      router = {
+        address = mkOption {
+          type = types.str;
+          description = lib.mdDoc "The IPv${toString v} address of the router.";
+        };
+      };
+
+      address = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = lib.mdDoc "The source IPv${toString v} address of the TAYGA server.";
+      };
+
+      pool = mkOption {
+        type = with types; nullOr (submodule (addrOpts v));
+        description = lib.mdDoc "The pool of IPv${toString v} addresses which are used for translation.";
+      };
+    };
+  };
+in
+{
+  options = {
+    services.tayga = {
+      enable = mkEnableOption (lib.mdDoc "Tayga");
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.tayga;
+        defaultText = lib.literalMD "pkgs.tayga";
+        description = lib.mdDoc "This option specifies the TAYGA package to use.";
+      };
+
+      ipv4 = mkOption {
+        type = types.submodule (versionOpts 4);
+        description = lib.mdDoc "IPv4-specific configuration.";
+        example = literalExpression ''
+          {
+            address = "192.0.2.0";
+            router = {
+              address = "192.0.2.1";
+            };
+            pool = {
+              address = "192.0.2.1";
+              prefixLength = 24;
+            };
+          }
+        '';
+      };
+
+      ipv6 = mkOption {
+        type = types.submodule (versionOpts 6);
+        description = lib.mdDoc "IPv6-specific configuration.";
+        example = literalExpression ''
+          {
+            address = "2001:db8::1";
+            router = {
+              address = "64:ff9b::1";
+            };
+            pool = {
+              address = "64:ff9b::";
+              prefixLength = 96;
+            };
+          }
+        '';
+      };
+
+      dataDir = mkOption {
+        type = types.path;
+        default = "/var/lib/tayga";
+        description = lib.mdDoc "Directory for persistent data";
+      };
+
+      tunDevice = mkOption {
+        type = types.str;
+        default = "nat64";
+        description = lib.mdDoc "Name of the nat64 tun device";
+      };
+
+      useNetworkd = mkOption {
+        type = types.bool;
+        default = false;
+        description = lib.mdDoc "Use systemd-networkd to configure the nat64 tun device";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    #networking.interfaces."${cfg.tunDevice}" = mkIf (!config.systemd.network.enable) {
```

Comment:
I would like to check if systemd-networkd is enabled as on other lines. But `mkIf (!config.systemd.network.enable)` runs into infinitie recursion.
```
georg@rick:~/git/nixpkgs/ > nom-build -j0 . -A nixosTests.tayga
/nix/store/23yy920js0pk5924c3899ajf7kxw7i7y-vm-test-run-tayga
Finished at 21:49:58 after 13s
georg@rick:~/git/nixpkgs/ > nom-build -j0 . -A nixosTests.tayga
error: infinite recursion encountered

       at /home/georg/git/nixpkgs/lib/modules.nix:528:12:

          527|         zipAttrsWith (n: concatLists)
          528|           (map (module: let subtree = module.${attr}; in
             |            ^
          529|               if !(builtins.isAttrs subtree) then
(use '--show-trace' to show detailed location information)
Finished at 21:50:06 after 1s
```